spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/ZuulConfiguration.java

```diff
@@ -179,8 +188,40 @@ public ZuulRouteApplicationContextInitializer zuulRoutesApplicationContextInitia
 		private Map<String, ZuulFilter> filters;
 
 		@Bean
-		public ZuulFilterInitializer zuulFilterInitializer() {
-			return new ZuulFilterInitializer(this.filters);
+		public ZuulFilterInitializer zuulFilterInitializer(
+				CounterFactory counterFactory, TracerFactory tracerFactory) {
+			FilterLoader filterLoader = FilterLoader.getInstance();
+			FilterRegistry filterRegistry = FilterRegistry.instance();
+			return new ZuulFilterInitializer(this.filters, counterFactory, tracerFactory, filterLoader, filterRegistry);
+		}
+
+	}
+
+	@Configuration
+	protected static class ZuulEmptyMetricsConfiguration {
```

Comment:
Moreover I will need to add `metrics-core` for `DropwizardAutoConfiguration` to be setup, because of this:
```
		@ConditionalOnBean(CounterService.class)
		@Bean
		public CounterFactory counterFactory(CounterService counterService) {
			return new DefaultCounterFactory(counterService);
		}
```
Maybe there exists some other way of conditional setup of these beans?